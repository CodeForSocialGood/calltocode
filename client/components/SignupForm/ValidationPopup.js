import React from 'react'
import PropTypes from 'prop-types'

import styles from './SignupForm.scss'

const getTickOrCross = (error, errorKey) => {
  const isError = error === null || error[errorKey]
  return isError ? styles.tick : styles.cross
}

const ValidationPopup = ({ error, active }) => {
  return (
    <div className={`${styles.validationPopup} ${active ? styles.show : styles.hide}` }>
      <p className={styles.bold}>Password must have</p>
      <ul>
        <li className={ getTickOrCross(error, 'upperCase') }> at least 1 UpperCase Character </li>
        <li className={ getTickOrCross(error, 'lowerCase') }> at least 1 LowerCase Character </li>
        <li className={ getTickOrCross(error, 'hasOneDigit') }> at least 1 Number </li>
        <li className={ getTickOrCross(error, 'hasSpecialChar') }> at least 1 Special Character </li>
        <li className={ getTickOrCross(error, 'minLength') }> at least 10 Characters </li>
        <li className={ getTickOrCross(error, 'maxLength') }> at most 128 Characters </li>
        <li className={ getTickOrCross(error, 'noIdenticalChars') }> not more than 2 identical characters in a row </li>
      </ul>
    </div>
  )
}

ValidationPopup.propTypes = {
  active: PropTypes.bool,
  error: PropTypes.object
}

export default ValidationPopup
