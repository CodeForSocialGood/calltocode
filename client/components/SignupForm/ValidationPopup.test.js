import React from 'react'
import test from 'ava'
import render from 'react-test-renderer'

import ValidationPopup from './ValidationPopup'

test('not active, errors all null', t => {
  const popup = render.create(<ValidationPopup error={null} active={false}/>).toJSON()
	t.snapshot(popup)
})

test.only('active, errors all null', t => {
  const popup = render.create(<ValidationPopup error={null} active={true}/>).toJSON()
	t.snapshot(popup)
})

test('active, errors all engaged', t => {
  const errors = {
    upperCase: true,
    lowerCase: true,
    hasOneDigit: true,
    hasSpecialChar: true,
    minLength: true,
    maxLength: true,
    noIdenticalChars: true
  }

  const popup = render.create(<ValidationPopup error={errors} active={true}/>).toJSON()
	t.snapshot(popup)
})

test('active, some errors engaged', t => {
  const errors = {
    upperCase: true,
    lowerCase: false,
    hasOneDigit: true,
    hasSpecialChar: false,
    minLength: true,
    maxLength: false,
    noIdenticalChars: true
  }

  const popup = render.create(<ValidationPopup error={errors} active={true}/>).toJSON()
	t.snapshot(popup)
})
