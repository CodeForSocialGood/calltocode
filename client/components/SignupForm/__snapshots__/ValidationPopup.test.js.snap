// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationPopup /> error is an object all errors falsey active is falsey component renders correctly 1`] = `
<div
  className="validationPopup hide"
>
  <p
    className="bold"
  >
    Password must have
  </p>
  <ul>
    <li
      className="cross"
    >
       at least 1 UpperCase Character 
    </li>
    <li
      className="cross"
    >
       at least 1 LowerCase Character 
    </li>
    <li
      className="cross"
    >
       at least 1 Number 
    </li>
    <li
      className="cross"
    >
       at least 1 Special Character 
    </li>
    <li
      className="cross"
    >
       at least 10 Characters 
    </li>
    <li
      className="cross"
    >
       at most 128 Characters 
    </li>
    <li
      className="cross"
    >
       not more than 2 identical characters in a row 
    </li>
  </ul>
</div>
`;

exports[`<ValidationPopup /> error is an object all errors falsey active is truthy component renders correctly 1`] = `
<div
  className="validationPopup show"
>
  <p
    className="bold"
  >
    Password must have
  </p>
  <ul>
    <li
      className="cross"
    >
       at least 1 UpperCase Character 
    </li>
    <li
      className="cross"
    >
       at least 1 LowerCase Character 
    </li>
    <li
      className="cross"
    >
       at least 1 Number 
    </li>
    <li
      className="cross"
    >
       at least 1 Special Character 
    </li>
    <li
      className="cross"
    >
       at least 10 Characters 
    </li>
    <li
      className="cross"
    >
       at most 128 Characters 
    </li>
    <li
      className="cross"
    >
       not more than 2 identical characters in a row 
    </li>
  </ul>
</div>
`;

exports[`<ValidationPopup /> error is an object all errors truthy active is falsey component renders correctly 1`] = `
<div
  className="validationPopup hide"
>
  <p
    className="bold"
  >
    Password must have
  </p>
  <ul>
    <li
      className="tick"
    >
       at least 1 UpperCase Character 
    </li>
    <li
      className="tick"
    >
       at least 1 LowerCase Character 
    </li>
    <li
      className="tick"
    >
       at least 1 Number 
    </li>
    <li
      className="tick"
    >
       at least 1 Special Character 
    </li>
    <li
      className="tick"
    >
       at least 10 Characters 
    </li>
    <li
      className="tick"
    >
       at most 128 Characters 
    </li>
    <li
      className="tick"
    >
       not more than 2 identical characters in a row 
    </li>
  </ul>
</div>
`;

exports[`<ValidationPopup /> error is an object all errors truthy active is truthy component renders correctly 1`] = `
<div
  className="validationPopup show"
>
  <p
    className="bold"
  >
    Password must have
  </p>
  <ul>
    <li
      className="tick"
    >
       at least 1 UpperCase Character 
    </li>
    <li
      className="tick"
    >
       at least 1 LowerCase Character 
    </li>
    <li
      className="tick"
    >
       at least 1 Number 
    </li>
    <li
      className="tick"
    >
       at least 1 Special Character 
    </li>
    <li
      className="tick"
    >
       at least 10 Characters 
    </li>
    <li
      className="tick"
    >
       at most 128 Characters 
    </li>
    <li
      className="tick"
    >
       not more than 2 identical characters in a row 
    </li>
  </ul>
</div>
`;
